packages/tre/src/plugins/cache/router.ts

```diff
@@ -1,40 +1,59 @@
-import { Response } from "undici";
+import { Request, RequestInit, Response } from "undici";
 import {
   DELETE,
   GET,
   PUT,
   RouteHandler,
   Router,
   decodePersist,
+  PURGE,
+  CfHeader,
 } from "../shared";
+import { CacheError, fallible } from "./errors";
 import { CacheGateway } from "./gateway";
 
 export interface CacheParams {
   namespace: string;
-  key: string;
+  uri: string;
 }
+
 export class CacheRouter extends Router<CacheGateway> {
-  @GET("/:namespace/:key")
+  @GET("/:uri")
   match: RouteHandler<CacheParams> = async (req, params) => {
+    const uri = decodeURIComponent(params.uri);
     const persist = decodePersist(req.headers);
-    const gateway = this.gatewayFactory.get(params.namespace, persist);
-    await gateway.match(params.key);
-    return new Response();
+    const ns = req.headers.get(CfHeader.CacheNamespace);
+    console.log(req.headers);
+    const gateway = this.gatewayFactory.get(
+      params.namespace + ns ? `:ns:${ns}` : `:default`,
+      persist
+    );
+    return fallible(gateway.match(new Request(uri, req as RequestInit)));
```

Comment:
I'm not sure I quite follow—key to the `fallible` method is calling `toResponse` on the error. How would `HttpError` help with that?

Reply:
The issue is that some of the errors need to have quite customised responses. An overridable method works though—I'll change to that. Since it touches R2 as well, I'll split it out into a separate PR though